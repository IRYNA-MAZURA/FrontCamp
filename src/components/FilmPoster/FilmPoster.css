.filmLink {
  text-decoration: none;
  max-width: 30%;
  padding: 0;
  margin: 0;
}

.film {
  display: flex;
  flex-direction: column;
  color: #555555;
  cursor: pointer;
  margin: 10px 20px;
}
  
.film-img {
  margin: 10px 20px;
  flex: 1;
  max-height: 650px;
  overflow: hidden;
}

.film-img img {
  width: 100%;
  height: auto;
  vertical-align: middle;
}

.film-info {
  display: flex;
  flex-direction: row;
  padding: 0 20px;
  margin-bottom: 10px;
}


.info-left {
  width: calc(100% - 80px);
  text-align: left;
}

.title {
  color: #FFFFFF;
}

.info-right {
  width: 80px;
  text-align: right;
}

.year {
  padding: 3px 10px;
  border: 1px solid #555555;
  border-radius: 3px;
  text-align: center;
}