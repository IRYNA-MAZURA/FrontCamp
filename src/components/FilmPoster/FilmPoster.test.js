import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { FilmPoster } from './FilmPoster';

const props = {
    film: {
        id: 1, 
        title: 'Harry Potter', 
        genres: ['fantasy'], 
        poster_path: 'https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.ebay.com.au%2Fitm%2FHARRY-POTTER-AND-THE-SORCERERS-STONE-MOVIE-POSTER-INTERNATIONAL-REGULAR-%2F122009198081&psig=AOvVaw1LDt0EqIHiVP4X7gHBXHY1&ust=1581665271313000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCJjn8beAzucCFQAAAAAdAAAAABAI', 
        release_date: '20/03/2000'
    }
};

describe('FilmPoster component', () => {
    it('snapshot testing', () => {
        const wrapper = renderer.create(<Router><FilmPoster {...props} /></Router>).toJSON();
        expect(wrapper).toMatchSnapshot();
    });
});