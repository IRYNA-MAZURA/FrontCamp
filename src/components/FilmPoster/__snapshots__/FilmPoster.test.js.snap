// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPoster component snapshot testing 1`] = `
<a
  className="filmLink"
  href="/movie/1"
  onClick={[Function]}
>
  <div
    className="film"
  >
    <div
      className="film-img"
    >
      <img
        alt=""
        src="https://www.google.com/url?sa=i&url=https%3A%2F%2Fwww.ebay.com.au%2Fitm%2FHARRY-POTTER-AND-THE-SORCERERS-STONE-MOVIE-POSTER-INTERNATIONAL-REGULAR-%2F122009198081&psig=AOvVaw1LDt0EqIHiVP4X7gHBXHY1&ust=1581665271313000&source=images&cd=vfe&ved=0CAIQjRxqFwoTCJjn8beAzucCFQAAAAAdAAAAABAI"
      />
    </div>
    <div
      className="film-info"
    >
      <div
        className="info-left"
      >
        <div
          className="title"
        >
          Harry Potter
        </div>
        <div
          className="genres"
        >
          fantasy
        </div>
      </div>
      <div
        className="info-right"
      >
        <div
          className="year"
        >
          20/03/2000
        </div>
      </div>
    </div>
  </div>
</a>
`;
